podo/PodoCount_code/podocount_main_serv.py

```diff
@@ -0,0 +1,298 @@
+"""
+Created on Mon Nov 23 9:00:07 2020
+@author: Briana Santo
+
+"""
+import numpy as np
+import pandas as pd
+import os
+import sys 
+sys.path.append('..')
+import time
+import warnings
+import argparse
+from PIL import Image
+from PodoCount_code.xml_to_mask import xml_to_mask
+from PodoCount_code.get_wsi_mask import get_wsi_mask
+from PodoCount_code.get_boundary import get_boundary
+from PodoCount_code.get_glom_props import get_glom_props
+from PodoCount_code.get_pod_feat_spaces import get_pod_feat_spaces
+from PodoCount_code.stain_decon import stain_decon
+from get_pod_props import get_pod_props
+from get_pod_props_mouse import get_pod_props_mouse
+from tiffslide import TiffSlide
+import girder_client
+from enum import Enum
+import shutil
+import json
+import xml.etree.ElementTree as ET
+from xml.dom import minidom
+from PodoCount_code.mask_to_xml import xml_create, xml_add_annotation, xml_add_region, xml_save
+from PodoCount_code.xml_to_mask_conversion import write_minmax_to_xml
+
+class InputType(Enum):
+    Mouse_Analysis = 'Mouse'
+    Human_Analysis = 'Human'
+
+#filter warnings
+warnings.filterwarnings("ignore")
+
+parser = argparse.ArgumentParser()
+parser.add_argument('--type')
+parser.add_argument('--input_image')
+#parser.add_argument('--glom_xml')
+parser.add_argument('--basedir')
+parser.add_argument('--slider')
+parser.add_argument('--section_thickness')
+parser.add_argument('--num_sections')
+parser.add_argument('--girderApiUrl')
+parser.add_argument('--girderToken')
+
+args = parser.parse_args()
+gc = girder_client.GirderClient(apiUrl=args.girderApiUrl)
+gc.setToken(args.girderToken)
+
+folder = args.basedir
+girder_folder_id = folder.split('/')[-2]
+file_name = args.input_image.split('/')[-1]
+files = list(gc.listItem(girder_folder_id))
+item_dict = dict()
+
+for file in files:
+    d = {file['name']: file['_id']}
+    item_dict.update(d)
+
+slide_item_id = item_dict[file_name]
+#get current working directory
+cwd = os.getcwd()
+WSIs = [args.input_image]
+
+#WSI general info
+section_thickness = int(args.section_thickness)
+
+NAMES = ['gloms']
+base_dir_id = folder.split('/')[-2]
+_ = os.system("printf '\nUsing data from girder_client Folder: {}\n'".format(folder))
+gc = girder_client.GirderClient(apiUrl=args.girderApiUrl)
+gc.setToken(args.girderToken)

```

Comment:
Why we authenticate girder client twice? let's keep the top one and remove this.